src/client/v2/algod/block.ts

```diff
@@ -21,9 +21,8 @@ export default class Block extends JSONRequest<BlockResponse, Uint8Array> {
 
   // eslint-disable-next-line class-methods-use-this
   prepare(body: Uint8Array): BlockResponse {
-    if (body && body.byteLength > 0) {
-      return BlockResponse.from_obj_for_encoding(encoding.decode(body));
-    }
-    return undefined;
+    return BlockResponse.from_obj_for_encoding(
```

Comment:
I actually removed the empty checking because I couldn't find any instance where it was used/required. If we get to this point, the response has a successful HTTP code, so I'd expect the body to always be populated. That should be true of the block endpoint and the few others I changed in this PR at least